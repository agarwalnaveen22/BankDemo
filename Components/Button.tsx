import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

const Button = (props: any) => {
  return (
    <TouchableOpacity
      onPress={props.onButtonPress}
      style={[styles.button, props.buttonStyle]}>
      <Text style={[styles.buttonText, props.buttonTextStyle]}>
        {props.buttonCaption}
      </Text>
    </TouchableOpacity>
  );
};

const styles = EStyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: '10rem',
  },
  buttonText: {
    fontSize: '20rem',
  },
});

export default Button;
