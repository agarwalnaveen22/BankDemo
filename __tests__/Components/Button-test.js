import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../Components/Button';

describe('test button container component', () => {
  it('match snapshot', () => {
    const snap = renderer.create(
    <Button 
        onButtonPress={() => jest.fn()}
        buttonCaption="Save"
    />
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});