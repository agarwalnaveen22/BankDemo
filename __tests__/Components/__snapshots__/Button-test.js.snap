// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button container component match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 19.736842105263158,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 39.473684210526315,
        },
        undefined,
      ]
    }
  >
    Save
  </Text>
</View>
`;
